ush/jjob_header.sh

```diff
@@ -87,28 +92,25 @@ source ./PDY
 export pid="${pid:-$$}"
 export pgmout="OUTPUT.${pid}"
 export pgmerr=errfile
+# TODO remove this when going to production
+# Needs to be set for err_chk/err_exit
+export pgm=${pgm:-}
 
 
 #############################
 # Source relevant config files
 #############################
 export EXPDIR="${EXPDIR:-${HOMEgfs}/parm/config}"
 for config in "${configs[@]:-''}"; do
-    source "${EXPDIR}/config.${config}"
-    status=$?
-    if (( status != 0 )); then
-        echo "FATAL [${BASH_SOURCE[0]}]: Unable to load config config.${config}"
-        exit "${status}"
-    fi
+    source "${EXPDIR}/config.${config}" && true
+    export err=$?
+    err_chk "FATAL ERROR [${BASH_SOURCE[0]}]: Unable to load config config.${config}"
 done
 
 
 ##########################################
 # Source machine runtime environment
 ##########################################
-source "${HOMEgfs}/env/${machine}.env" "${env_job}"
-status=$?
-if (( status != 0 )); then
-    echo "FATAL [${BASH_SOURCE[0]}]: Error while sourcing machine environment ${machine}.env for job ${env_job}"
-    exit "${status}"
-fi
+source "${HOMEgfs}/env/${machine}.env" "${env_job}" && true
+export err=$?
+err_chk "FATAL ERROR: [${BASH_SOURCE[0]}]: Error while sourcing machine environment ${machine}.env for job ${env_job}"
```

Comment:
How does this work without an if-block wrapped around it? Won't it always invoke `err_chk` with the provided input message?

Reply:
`err_chk` has an if block inside it that checks `err`.  If it is non-zero, then it echoes the `FATAL ERROR` message.

Reply:
Gotcha, thanks!